src/PolyhedralGeometry/Polyhedron/standard_constructions.jl

```diff
@@ -1233,3 +1205,1108 @@ function rand_subpolytope(P::Polyhedron{T}, n::Int; seed=nothing) where T<:scala
   pm_obj = Polymake.polytope.Polytope{_scalar_type_to_polymake(T)}(VERTICES=pm_matrix)::Polymake.BigObject
   return Polyhedron{T}(pm_obj, coefficient_field(P))
 end
+
+@doc raw"""
+    SIM_body_polytope(alpha::AbstractVector)
+
+Produce an $n$-dimensional SIM-body as generalized permutahedron in 
+$(n+1)$-space. SIM-bodies are defined in [GK14](@cite), but the input needs to 
+be descending instead of ascending, as used in [JKS22](@cite), i.e. `alpha` has parameters
+$(a_1,\dots,a_n)$ such that $a_1 \geq \dots \geq a_n \geq 0$.  
+
+# Example
+To produce a $2$-dimensional SIM-body, use for example the following code. 
+Note that the polytope lives in $3$-space, so we project it down to $2$-space 
+by eliminating the last coordinate. 
+
+```jldoctest
+julia> s = SIM_body_polytope([3,1])
+Polyhedron in ambient dimension 3
+
+julia> p = convex_hull(map(x->x[1:dim(s)],vertices(s)))
+Polyhedron in ambient dimension 2
+
+julia> vertices(p) 
+5-element SubObjectIterator{PointVector{QQFieldElem}}:
+ [0, 0]
+ [3, 0]
+ [3, 1]
+ [0, 3]
+ [1, 3]
+```
+"""
+function SIM_body_polytope(alpha::Vector) 
+    n = length(alpha)
+    if n < 1
+        throw(ArgumentError("SIM_body_polytope: dimension must be at least 1"))
+    end
+    for i in 1:n-1
+        if alpha[i]<alpha[i+1]
+            throw(ArgumentError("SIM_body_polytope: input is not descending"))
+        end
+    end
```

Comment:
```suggestion
    @req n >= 1 "Dimension must be at least 1"
    for i in 1:n-1
        @req alpha[i] >= alpha[i+1] "Input is not descending"
    end
```
In OSCAR style, we would use the `@req` macro instead of throwing errors. Furthermore, the name of the function throwing some error needn't be mentioned in the error message as it is already included in the error stack.
This pattern could be applied to many places in your PR.